src/features/EditMixedValues.cpp

```diff
@@ -151,18 +153,9 @@ class MixedValuesInput : public CCMenu {
         }
         this->updateLabel();
     }
-    void onNextFree(CCObject*) {
-        std::set<T> usedLayers;
-        for (auto obj : CCArrayExt<GameObject*>(LevelEditorLayer::get()->m_objects)) {
-            m_config.useNextFreeValues(obj, usedLayers);
-        }
-        T nextFree;
-        for (nextFree = m_config.limits.min; nextFree < m_config.limits.max; nextFree += 1) {
-            if (!usedLayers.contains(nextFree)) {
-                break;
-            }
-        }
-        this->override(nextFree);
+    void onNextFree(CCObject* sender) {
+        // this exists trust
+        (static_cast<SetGroupIDLayer*>(CCDirector::get()->getRunningScene()->getChildByID("SetGroupIDLayer"))->*m_config.nextFreeFunction)(sender);

```

Comment:
Realistically `m_nextFreeFunction` should be an `std::function<void()>` that captures the `SetGroupIDLayer` when the `MixedValuesInput` is created rather than whatever this is